### PR TITLE
Feature/Add url encoding

### PR DESCRIPTION
**Pull request by Miroshni4enko**

### CODE REVIEW

File: src/main/java/org/truemail_rb/TruemailClient.java [hunk 1]
```diff
@@ -68,7 +70,10 @@ private String uri(String email) {
         result.append("?email=");
         result.append(email);
 
-        return result.toString();
+        // This way of encoding looks strange, but there is no other way in vanila java
+        // See https://stackoverflow.com/questions/444112/how-do-i-encode-uri-parameter-values
+        return URLEncoder.encode(result.toString(), "UTF-8")
```
Comment (bestwebua) [inline src/main/java/org/truemail_rb/TruemailClient.java:75]: Should be encoded just uri params part.

File: src/test/java/org/truemail_rb/TruemailClientTests.java [hunk 2]
```diff
@@ -23,7 +23,7 @@
     @Test
     public void validateReturnsSuccessResponseWhenValidEmail() throws Exception {
         URL u = mock(URL.class);
-        String url = "http://localhost:9292?email=test.email@google.com";
+        String url = "http%3A%2F%2Flocalhost%3A9292%3Femail%3Dtest.email%40google.com";
```
Comment (bestwebua) [inline src/test/java/org/truemail_rb/TruemailClientTests.java:26]: The same.

File: src/test/java/org/truemail_rb/TruemailClientTests.java [hunk 3]
```diff
@@ -47,7 +47,7 @@ public void validateReturnsSuccessResponseWhenValidEmail() throws Exception {
     @Test
     public void validateReturnsErrorResponseWhenHttpConnectionFailed() throws Exception {
         URL u = mock(URL.class);
-        String url = "https://test.host:12345?email=test.email";
+        String url = "https%3A%2F%2Ftest.host%3A12345%3Femail%3Dtest.email";
```
Comment (bestwebua) [inline src/test/java/org/truemail_rb/TruemailClientTests.java:50]: And all below.

File: src/main/java/org/truemail_rb/TruemailClient.java (hunk 1 above)
Comment (Miroshni4enko) [reply src/main/java/org/truemail_rb/TruemailClient.java]: Done)

File: src/test/java/org/truemail_rb/TruemailClientTests.java (hunk 2 above)
Comment (Miroshni4enko) [reply src/test/java/org/truemail_rb/TruemailClientTests.java]: Done)

File: src/test/java/org/truemail_rb/TruemailClientTests.java (hunk 3 above)
Comment (Miroshni4enko) [reply src/test/java/org/truemail_rb/TruemailClientTests.java]: Done)

File: src/test/java/org/truemail_rb/TruemailClientTests.java [hunk 4]
```diff
@@ -23,7 +23,7 @@
     @Test
     public void validateReturnsSuccessResponseWhenValidEmail() throws Exception {
         URL u = mock(URL.class);
-        String url = "http://localhost:9292?email=test.email@google.com";
+        String url = "http://localhost:9292?email=test.email%40google.com";
```
Comment (bestwebua) [inline src/test/java/org/truemail_rb/TruemailClientTests.java:26]: 👍

File: src/main/java/org/truemail_rb/TruemailClient.java [hunk 5]
```diff
@@ -57,18 +59,26 @@ private String getResponseBody(InputStream inputStream) throws IOException {
...
+        path.append(this.truemailConfiguration.getHost());
+        path.append(":");
+        path.append(this.truemailConfiguration.getPort());
+        return path.toString();
+    }
 
-        return result.toString();
+    private String getEncodedParams(String email) throws UnsupportedEncodingException {
+        // This way of encoding looks strange, but there is no other way in vanilla java
+        // See https://stackoverflow.com/questions/444112/how-do-i-encode-uri-parameter-values
```
Comment (bestwebua) [inline src/main/java/org/truemail_rb/TruemailClient.java:79]: Maybe this reference to stackoverflow is bit excessive?

**Review by bestwebua — APPROVED** [review]

**Pull request merged**